### DISCUSSION TITLE
fs readdir works in dev, not on vercel

### DISCUSSION THREAD

**Discussion by Glutch** [vercel/next.js#34295] [Help]
Hello!

This code works perfectly in development, but when i upload to production at vercel i get
`[Error: ENOENT: no such file or directory, scandir '/public/testfolder']`

```
export async function getServerSideProps(context) {
  const user = await getSession(context)
  
  if (!user) {
    return {
      props: {user: false},
    }
  }
  
  const files = await readdir('/public/testfolder')
  return {
    props: {user, files},
  }
}
```

I've read multiple threads about this but no one finds a solution, and i've seen no clear answer
https://stackoverflow.com/questions/70027742/how-to-scan-directories-on-production-on-vercel-with-next-js

**Comment by GabenGar** [comment]
You need to normalize the path to the public folder, because currently it looks for `/public` at the root of your machine. Which might be the case with dev setup, but clearly it's not in the prod.
Something along these lines:
```js
const testPath = path.resolve(process.cwd(), "public", "testfolder");
const files = await readdir(testPath)
```

**Reply by Glutch** [reply]
Thank you but i can't seem to get it to work
I've tried this and nothing works.
If i browse the source -> output tab on vercel, i can clearly see that my folders are right there next to 'api' and '_next' folders

```
path.resolve('./test')
path.resolve('test')
path.resolve('public', 'test')
path.resolve('./public', 'test')
path.join('./test')
path.join('test')
path.join('public', 'test')
path.join('./public', 'test')
path.join(process.cwd(), './test')
path.join(process.cwd(), 'test')
path.join(process.cwd(), 'public', 'test')
path.join(process.cwd(), './public', 'test')
path.resolve(process.cwd(), './test')
path.resolve(process.cwd(), 'test')
path.resolve(process.cwd(), 'public', 'test')
path.resolve(process.cwd(), './public', 'test')
```

**Reply by GabenGar** [reply]
Try to use the folder other than `public`, because it gets transformed in the build process.

**Reply by Glutch** [reply]
`const files = await readdir(path.resolve(process.cwd(), 'test', 'protocol'))`

```
[Error: ENOENT: no such file or directory, scandir '/var/task/test/protocol'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'scandir',
  path: '/var/task/test/protocol'
}
```

Same error. I'm trying to create a simple solution to just drop some pdf's into a folder (public/protocol) then readdir, and link to them on the site. I believe this would not work if i used another folder than public. This is a simple solution for a small community, so i'm not looking into AWS S3 or something similar

**Reply by GabenGar** [reply]
Pretty strange it doesn't work, since [the blog example creates the path the same way](https://nextjs.org/learn/basics/data-fetching/implement-getstaticprops).

**Reply by Glutch** [reply]
I think the difference is that the blog example reads a folder from /pages and not /public
I've been trying the whole day to solve this problem, super annoying :(

**Reply by Glutch** [reply]
Update: adding webpack5: false, in next.config.js made `await readdir(path.resolve(process.cwd(), 'test', 'protokoll'))` this work.
However, now i need a duplicate of that folder in both test (for the server to read the files) and in public for the user to access them...

**Comment by Andrehatlo** [comment]
Still no updates for this?

**Comment by ba-ppp** [comment]
I resolved it by add it `path.resolve(process.cwd(), "your folder")`
This means you will figure out the current path of your source, not the built source

**Comment by Baltazarka** [comment]
I have this problem too.
![image](https://github.com/vercel/next.js/assets/50296997/7d5f9d03-5daf-40a2-ac39-b307c0e5266e)
![image](https://github.com/vercel/next.js/assets/50296997/4c8da15e-a290-464b-933a-180d94f86472)
